internal/escape/escape.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2019 The Go Cloud Development Kit Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package escape includes helpers for escaping and unescaping strings.
+package escape
+
+// Escape returns s, with all bytes for which shouldEscape returns true
```

Comment:
What's going to happen if the input is non-ASCII UTF-8? You should at least talk about that somewhere.

Reply:
Does it matter what the input is? We're currently going through it byte by byte, so there's no interpretation of UTF-8 going on. Do you think it would be better to have `shouldEscape` take a rune, and iterate over the `s` using `range` so that we go rune-by-rune instead of byte-by-byte?

Reply:
WDYT about having `shouldEscape` take a slice of `rune` instead (and the index)? If it returns true, we could still escape the bytes independently, or do the whole rune.

Reply:
I've done this, PTAL.